src/Products/SiteErrorLog/SiteErrorLog.py

```diff
@@ -329,3 +332,18 @@ def manage_addErrorLog(dispatcher, RESPONSE=None):
         RESPONSE.redirect(
             dispatcher.DestinationURL() +
             '/manage_main?manage_tabs_message=Error+Log+Added.')
+
+@adapter(IPubFailure)
+def _handle_wsgi_publisher_failure(event):
+    published = event.request.get('PUBLISHED')
+    if not published:
+        return
+
+    try:
+        error_log = aq_acquire(published, '__error_log__', containment=1)
+    except AttributeError:
+        pass
+    else:
+        error_log.raising(event.exc_info)
+
+provideHandler(_handle_wsgi_publisher_failure)
```

Comment:
Registering the handler should be done via ZCML in `configure.zcml` so it is registered in the correct registry.

See https://github.com/zopefoundation/zope.catalog/blob/ccbd0f444a0dd69d4fa244741215f43e9ef46648/src/zope/catalog/subscribers.zcml as an example.